File: common/common-core/src/main/java/io/vanaheimr/common/core/RequestContext.java
package io.vanaheimr.common.core;


import io.micronaut.http.HttpRequest;

/**
 * A class that holds the current request context.
 */
public class RequestContext {

    private HttpRequest request;

    private String userId;

    private String tenantId;

    private String requestId;

    public RequestContext() {

    }


    public HttpRequest getRequest() {
        return request;
    }

    public void setRequest(HttpRequest request) {
        this.request = request;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getTenantId() {
        return tenantId;
    }

    public void setTenantId(String tenantId) {
        this.tenantId = tenantId;
    }

    public String getRequestId() {
        return requestId;
    }

    public void setRequestId(String requestId) {
        this.requestId = requestId;
    }


    @Override
    public String toString() {
        return "RequestContext{" +
                "request=" + request +
                ", userId='" + userId + '\'' +
                ", tenantId='" + tenantId + '\'' +
                ", requestId='" + requestId + '\'' +
                '}';
    }
}

File: common/common-core/src/main/kotlin/io/vanaheimr/common/core/RequestContext.kt
package io.vanaheimr.common.core

import io.micronaut.http.HttpRequest
import kotlin.coroutines.AbstractCoroutineContextElement
import kotlin.coroutines.CoroutineContext

/**
 * A class that holds the current request context.
 */
class RequestContext(private val map: MutableMap<String, Any> = HashMap(56)) : MutableMap<String, Any> by map,
    AbstractCoroutineContextElement(RequestContext) {
    var userId: Long by this

    var userName: String by this

    var roleIds: List<Long> by this

    var permissionIds: List<Long> by this

    var request: HttpRequest<*> by this

    // 租户id
    var tenantId: Long by this

    companion object : CoroutineContext.Key<RequestContext>
}
